### PR TITLE
11879 warning for func package patterns

### PR DESCRIPTION
**Pull request by Lokesh-Jawale**
Closes: [#11879](https://github.com/serverless/serverless/issues/11879)

### CODE REVIEW

**Review by medikoo — CHANGES_REQUESTED** [review]
@Lokesh-Jawale great thanks for giving it a spin

In general it looks good, however I think more appropriate place for this warning will be the point where were resolve the packaging rules (and package functions), so here: https://github.com/serverless/serverless/blob/ed15cb27aee68954c93d875da96274914943ad71/lib/plugins/package/lib/package-service.js#L101

A place where you've added the logic is responsible for compiling AWS CloudFormation template, so it's not that related to packaging

File: lib/plugins/package/lib/package-service.js [hunk 1]
```diff
@@ -99,6 +100,25 @@ module.exports = {
         await this.packageFunction(functionName);
         return;
       }
+
+      // show warning if package patterns in function are provided
+      // without package.individually set to true at service level or function level.
+      if (
+        functionObject.package.patterns &&
+        !functionObject.package.individually &&
+        !this.serverless.service.package.individually &&
```
Comment (medikoo) [inline lib/plugins/package/lib/package-service.js:109]: Let's remove those two lines, we're already in context where that is confirmed

File: lib/plugins/package/lib/package-service.js [hunk 2]
```diff
@@ -99,6 +100,25 @@ module.exports = {
...
+        !shownFuncPackageWarning
+      ) {
+        shownFuncPackageWarning = true;
+        log.warning();
+        const warningMessage = [
+          'Package patterns at function level are only applicable ',
+          'if package.individually is set to true at service level or function level in serverless.yaml.',
+          ' Serverless will ignore the patterns and the default packaging patterns will be applied.',
```
Comment (medikoo) [inline lib/plugins/package/lib/package-service.js:117]: Let's maybe reword last line to:

```
The framework will ignore the patterns defined at the function level and apply only the service-wide ones
```

**Review by medikoo — CHANGES_REQUESTED** [review]
@Lokesh-Jawale that looks very good. I've just proposed some small improvements

**Comment by Lokesh-Jawale** [comment]
@medikoo thanks for the suggestions, I have updated the PR as per the suggestions.

**Review by medikoo — APPROVED** [review]
Thank you 👍
